sdk/test/Services/DynamoDBv2/UnitTests/Custom/ExpressionBuilderTests.cs

```diff
@@ -0,0 +1,434 @@
+            Assert.AreEqual(4, updateExpression.ExpressionAttributeValues.Count);
+        }
+
+        [TestMethod]
+        [TestCategory("DynamoDBv2")]
+        public void UpdateExpressionBuilder_SetAndRemoveExpressionBuilder_Build_Correctly()
+        {
+            var updateExpression = UpdateExpressionBuilder.New().
+                Set(NameBuilder.New("test"),
+                    SetValueBuilder.New().WithValue(10).Plus(20)).
+                Remove(NameBuilder.New("test2")).
+                Build();
+
+            Assert.AreEqual("SET #U0 = :U0 + :U1\nREMOVE #U1\n" ,updateExpression.ExpressionStatement);
+            Assert.AreEqual(2, updateExpression.ExpressionAttributeNames.Count);
+            Assert.AreEqual(2, updateExpression.ExpressionAttributeValues.Count);
+        }
+
+        [TestMethod]
+        [TestCategory("DynamoDBv2")]
+        public void ConditionExpressionBuilderTest()
+        {
+            var conditionExpression = ConditionExpressionBuilder.New().WithName("Age").Equal(21).
+                And(ConditionExpressionBuilder.New().WithName("Status").GreaterThan(1)).Build();
+
+            Assert.AreEqual("(#C0 = :C0) AND (#C1 > :C1)", conditionExpression.ExpressionStatement);
+            Assert.AreEqual(2, conditionExpression.ExpressionAttributeNames.Count);
+            Assert.AreEqual("Age", conditionExpression.ExpressionAttributeNames["#C0"]);
+            Assert.AreEqual("Status", conditionExpression.ExpressionAttributeNames["#C1"]);
+            Assert.AreEqual(2, conditionExpression.ExpressionAttributeValues.Count);
+        }
+
+        [TestMethod]
+        [TestCategory("DynamoDBv2")]
+        public void NameBuilderConstructor_WhenCalled_CreatesValidInstance()
+        {
+            var nameBuilder = NameBuilder.New("test");
+
+            Assert.IsNotNull(nameBuilder);
+            Assert.IsInstanceOfType(nameBuilder, typeof(OperandBuilder));
+        }
+
+        [TestMethod]
+        [TestCategory("DynamoDBv2")]
+        public void Complex_Attribute_Paths_Should_Build_Correctly()
+        {
+            var nameBuilder = NameBuilder.New("parent.child[0].attribute");
+            var node = nameBuilder.AttributeExists().Build();
+
+            Assert.AreEqual("attribute_exists (#C0.#C1[0].#C2)", node.ExpressionStatement);
+            Assert.AreEqual(0, node.ExpressionAttributeValues.Count);
+            Assert.AreEqual(3, node.ExpressionAttributeNames.Count);
+        }
+
+        [TestMethod]
+        [TestCategory("DynamoDBv2")]
+        public void BuildNameBuilder_WithSpecialCharacters_Should_Build_Correctly()
+        {
+            var attributeName = "Test#Attribute-Name.123";
+            var nameBuilder = NameBuilder.New(attributeName);
+
+            var result = nameBuilder.AttributeExists().Build();
+
+            Assert.IsNotNull(result);
+            Assert.AreEqual("attribute_exists (#C0.#C1)", result.ExpressionStatement);
```

Comment:
Can we make sure every test is also asserting the `ExpressionAttributeNames` and `ExpressionAttributeValues`. And not just the count but the values in the collection.